In [42]:
import numpy as np
import pandas as pd
import  scipy.stats                     as  stats
import  statsmodels.stats.proportion    as  smp
import matplotlib.pyplot     as     plt
import seaborn as sns
import statsmodels.api         as     sm
from   statsmodels.formula.api import ols
from scipy.stats             import ttest_1samp,ttest_ind, wilcoxon
from statsmodels.stats.power import ttest_power

In [43]:
old_sch=np.array([57000,103000,59000,75000,84000,73000,35000,110000,44000,82000,67000,64000,78000,53000,41000,39000,80000,87000,73000,65000,28000,62000,49000,84000,63000,77000,67000,101000,91000,50000])
new_sch=np.array([62000,122000,54000,82000,84000,86000,32000,104000,38000,107000,84000,85000,99000,39000,34000,58000,73000,53000,66000,78000,41000,71000,38000,95000,81000,58000,75000,94000,100000,68000])


In [44]:
#the mean of the old_scheme:
olds=old_sch.mean()

print('The mean of the old_scheme is %1.3f'% olds)

The mean of the old_scheme is 68033.333


In [45]:
# the mean of the new_scheme:
nesc=new_sch.mean()
print('The mean of the new_scheme is %1.3f'%nesc)

The mean of the new_scheme is 72033.333


In [70]:
######2.	Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs

# Step 1: Define null and alternative hypothesis

In testing that the new scheme has incresed effect over old scheme, the null hypothesis states that difference in scheme, $\mu{New}$ equals $\mu{Old}$. The alternative hypthesis states that difference in scheme are not equal, $\mu{New}$ $\neq$ $\mu{Old}$

* $H_0$: $\mu{New}$ - $\mu{Old}$ =  0
* $H_A$: $\mu{New}$ - $\mu{Old}$ $\neq$  0


# Step 2: Decide the significance level

Here we select 
α
α
= 0.05 and sample size < 30 and population standard deviation is not known.

# Step 3: Identify the test statistic

* Sample sizes for both samples are  same.
* We have two paired samples and we do not know the population standard deviation.
* The sample is not a large sample, n =30. So you use the t distribution and the $t_STAT$ test statistic for two sample paired test.

# Step 4: Calculate the p - value and test statistic

Here we give the two sample observations as input. This function returns t statistic and two-tailed p value.**

In [60]:
t_statistic, p_value  =  stats.ttest_rel(new_sch, old_sch )
print('P Value %1.3f' % p_value)  
print('t_statistic %1.3f'%t_statistic)

P Value 0.131
t_statistic 1.556


# Step 5: Decide to reject or accept null hypothesis

In this example, p value is 0.131 and it is greater than 5% level of significance

So the statistical decision is that we fail to reject the null hypothesis at 5% level of significance.

### So there is  insufficient evidence  to reject the null hypothesis that the new scheme has significantly raised outputs

In [68]:
#####   The probability of a type 1 error

In [62]:
n         = 30    # sample size
sigma     = new_sch.std()  # population standard deviation
serr      = sigma / np.sqrt(n) # Standard Error

alpha     = 0.05     # significance level
mu0       = 5000 #  hypothetical lower bound
q         = int(round(stats.norm.isf(q =  alpha, loc = mu0, scale = serr),0))
print(q)

12105


* Assume actual mean new_sch is 72033.333 
* We need to find the P(Population mean $\<$ 72033.333  | $H_0$ is true)

In [69]:
mu1  = 72033.333 # Actual mean

p = round(1 - stats.norm.cdf(q, loc = mu1, scale = serr),4)
print('At 5 %s level of significance, the probability of having type I error\n\
       for a sample size of 30  is %2.4f' %('%',p))


At 5 % level of significance, the probability of having type I error
       for a sample size of 30  is 1.0000
